### PR TITLE
updating demo files & scripts

### PR DESCRIPTION
**Pull request by shawnmcclelland**
Fixed a bug where looking around would cause the player transform to drift and added a new input action to handle physics based jumping.

### CODE REVIEW

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs [hunk 1]
```diff
@@ -47,9 +63,17 @@ public void Awake()
...
+            if(isGrounded == true)
+            {
+                rb.AddForce(m_Jump * jumpForce, ForceMode.Impulse);
+                isGrounded = false;
+            }
+        };
     }
 
-    public void OnEnable()
+	public void OnEnable()
```
Comment (Rene-Damm) [inline Assets/Demo/SimpleController_UsingActions_InAsset.cs]: Please run format.ps1/format.sh (found in repo root).

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs [hunk 2]
```diff
@@ -80,10 +104,20 @@ private void Move(Vector2 direction)
...
     private void Look(Vector2 rotate)
     {
-        var scaledRoateSpeed = rotateSpeed * Time.deltaTime;
-        m_Rotation.y += rotate.x * scaledRoateSpeed;
-        m_Rotation.x = Mathf.Clamp(m_Rotation.x - rotate.y * scaledRoateSpeed, -89, 89);
-        transform.localEulerAngles = m_Rotation;
+        //var scaledRoateSpeed = rotateSpeed * Time.deltaTime;
+        //m_Rotation.y += rotate.x * scaledRoateSpeed;
+        //m_Rotation.x = Mathf.Clamp(m_Rotation.x - rotate.y * scaledRoateSpeed, -89, 89);
+        //transform.localEulerAngles = m_Rotation;
```
Comment (Rene-Damm) [inline Assets/Demo/SimpleController_UsingActions_InAsset.cs]: No commented out code without cause please. Either add comment why code is being kept or remove commented out code.

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs [hunk 3]
```diff
@@ -8,16 +8,32 @@ public class SimpleController_UsingActions_InAsset : MonoBehaviour
...
 
     private Vector2 m_Move;
     private Vector2 m_Look;
+    private Vector3 m_Jump;
+    private bool isGrounded;
     private bool m_Charging;
 
     private Vector2 m_Rotation;
 
+    Rigidbody rb;
```
Comment (Rene-Damm) [inline Assets/Demo/SimpleController_UsingActions_InAsset.cs]: `m_RigidBody`.

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs [hunk 4]
```diff
@@ -8,16 +8,32 @@ public class SimpleController_UsingActions_InAsset : MonoBehaviour
...
     public float rotateSpeed;
     public float burstSpeed;
+    public float jumpForce = 2.0f;
     public GameObject projectile;
 
     public DemoControls controls;
 
     private Vector2 m_Move;
     private Vector2 m_Look;
+    private Vector3 m_Jump;
```
Comment (Rene-Damm) [inline Assets/Demo/SimpleController_UsingActions_InAsset.cs]: No need to have this one.

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs [hunk 5]
```diff
@@ -80,10 +104,20 @@ private void Move(Vector2 direction)
...
+        //transform.localEulerAngles = m_Rotation;
+
+        var clampAngle = 80.0f;
+
+        m_Rotation.y += rotate.x * rotateSpeed * Time.deltaTime;
+        m_Rotation.x += rotate.y * rotateSpeed * Time.deltaTime;
+
+        m_Rotation.x = Mathf.Clamp(m_Rotation.x, -clampAngle, clampAngle);
+
+        Quaternion localRotation = Quaternion.Euler(m_Rotation.x, m_Rotation.y, 0.0f);
```
Comment (Rene-Damm) [inline Assets/Demo/SimpleController_UsingActions_InAsset.cs]: Use var.

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs [hunk 6]
```diff
@@ -47,9 +63,17 @@ public void Awake()
         {
             m_Charging = false;
         };
+        controls.gameplay.jump.performed += ctx =>
+        {
+            if(isGrounded == true)
```
Comment (Rene-Damm) [inline Assets/Demo/SimpleController_UsingActions_InAsset.cs]: `if (isGrounded)`

**Review by Rene-Damm — CHANGES_REQUESTED** [review]

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs (hunk 4 above)
Comment (shawnmcclelland) [reply Assets/Demo/SimpleController_UsingActions_InAsset.cs]: I could remove the private var and just have a var declaration inside of Awake(). m_Jump is re-used on 28 and 69 but that could get consolidated into a single var within the Awake loop I think.

Comment (Rene-Damm) [reply Assets/Demo/SimpleController_UsingActions_InAsset.cs]: 29 is just an assignment and 70 the only read it seems. Doesn't seem like it needs a var outside of the jump response code.

File: Assets/Demo/SimpleController_UsingActions_InAsset.cs [hunk 7]
```diff
@@ -80,10 +103,15 @@ private void Move(Vector2 direction)
...
-        transform.localEulerAngles = m_Rotation;
+        var clampAngle = 80.0f;
+
+        m_Rotation.y += rotate.x * rotateSpeed * Time.deltaTime;
+        m_Rotation.x += rotate.y * rotateSpeed * Time.deltaTime;
+
+        m_Rotation.x = Mathf.Clamp(m_Rotation.x, -clampAngle, clampAngle);
+
+        var localRotation = Quaternion.Euler(m_Rotation.x, m_Rotation.y, 0.0f);
+        transform.rotation = localRotation;
```
Comment (Rene-Damm) [inline Assets/Demo/SimpleController_UsingActions_InAsset.cs:114]: Inverts Y.

File: Assets/Demo/Demo.unity [hunk 8]
```diff
@@ -539,6 +570,94 @@ Transform:
   m_Father: {fileID: 0}
   m_RootOrder: 2
   m_LocalEulerAnglesHint: {x: 0, y: 0, z: 0}
+--- !u!1 &1065433050
+GameObject:
```
Comment (Rene-Damm) [inline Assets/Demo/Demo.unity:574]: Didn't catch this before. Shouldn't be in here.

File: Assets/Demo/Demo.unity [hunk 9]
```diff
@@ -367,6 +370,34 @@ Transform:
...
+  m_GameObject: {fileID: 919824548}
+  m_Material: {fileID: 0}
+  m_IsTrigger: 0
+  m_Enabled: 1
+  m_Radius: 0.5
+  m_Height: 2
+  m_Direction: 1
+  m_Center: {x: 0, y: 0, z: 0}
+--- !u!54 &919824555
+Rigidbody:
```
Comment (Rene-Damm) [inline Assets/Demo/Demo.unity:387]: Physics-based movement updates are now fighting direct transform updates.

**Pull request merged**